### PR TITLE
Put catch in place for nsfw unexpanded posts to display placeholder

### PR DESCRIPTION
**Pull request by danehansen**
:eyeglasses: @ajacksified or @curioussavage

### CODE REVIEW

File: src/views/components/ListingContent.jsx [hunk 1]
```diff
@@ -101,7 +101,9 @@ class ListingContent extends React.Component {
     var preview = this._previewImageUrl(expanded);
 
     if (media && oembed) {
-      var thumbnailUrl = oembed.thumbnail_url;
+      if (expanded || !ListingContent.isNSFW(listing)) {
```
Comment (ajacksified) [inline src/views/components/ListingContent.jsx]: Checking `_isExpanded` will embed nsfw thumbs on a comment page. This should instead check `state.expanded`, to catch if it had a placeholder on a comment page, and then was explicitly expanded.

File: src/views/components/ListingContent.jsx [hunk 2]
```diff
@@ -390,12 +396,14 @@ class ListingContent extends React.Component {
...
           if (source) {
-            return _limitAspectRatio(source.width / source.height);
+            ratio = source.width / source.height;
+            return props.single ? ratio : _limitAspectRatio(ratio);
           }
         }
       }
     }
 
+    // console.log('ListingContent._aspectRatio: missed a case', listing);
```
Comment (ajacksified) [inline src/views/components/ListingContent.jsx]: nuke

**Comment by ajacksified** [comment]
:nail_care:

**Comment by danehansen** [comment]
:haircut: @ajacksified

File: src/views/components/ListingContent.jsx [hunk 3]
```diff
@@ -390,12 +396,17 @@ class ListingContent extends React.Component {
...
+            return props.single ? ratio : _limitAspectRatio(ratio);
           }
         }
       }
     }
 
+    //TODO: this doesn't work
+    /*if (props.app.config.debug) {
+      console.log('ListingContent._aspectRatio: missed a case', listing);
+    }*/
```
Comment (ajacksified) [inline src/views/components/ListingContent.jsx]: nuke

File: src/views/components/ListingContent.jsx [hunk 4]
```diff
@@ -101,7 +101,9 @@ class ListingContent extends React.Component {
     var preview = this._previewImageUrl(expanded);
 
     if (media && oembed) {
-      var thumbnailUrl = oembed.thumbnail_url;
+      if (state.expanded || !ListingContent.isNSFW(listing)) {
```
Comment (ajacksified) [inline src/views/components/ListingContent.jsx]: `state` is undefined

**Comment by ajacksified** [comment]
:nail_care:

**Comment by ajacksified** [comment]
@danehansen the case where a post has an embed thumbnail, but no preview, isn't solved by this. Example: https://m.reddit.com/r/Unexpected/comments/3awtgs/the_proper_use_of_a_banana/

**Pull request merged**